src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.AspNet.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Hosting.Internal
+{
+    internal static class HostingLoggerExtensions
+    {
+        public static IDisposable RequestScope(this ILogger logger, HttpContext httpContext)
+        {
+            return logger.BeginScopeImpl(new HostingLogScope(httpContext));
+        }
+
+        public static void RequestStarting(this ILogger logger, HttpContext httpContext)
+        {
+            if (logger.IsEnabled(LogLevel.Information))
+            {
+                logger.Log(
+                    logLevel: LogLevel.Information,
+                    eventId: 1,
+                    state: new HostingRequestStarting(httpContext),
+                    exception: null,
+                    formatter: HostingRequestStarting.Callback);
+            }
+        }
+
+        public static void RequestFinished(this ILogger logger, HttpContext httpContext)
+        {
+            if (logger.IsEnabled(LogLevel.Information))
+            {
+                logger.Log(
+                    logLevel: LogLevel.Information,
+                    eventId: 2,
+                    state: new HostingRequestFinished(httpContext),
+                    exception: null,
+                    formatter: HostingRequestFinished.Callback);
+            }
+        }
+
+        private class HostingLogScope : ILogValues
+        {
+            private readonly HttpContext _httpContext;
```

Comment:
Could move this up to ```HttpAbstractions``` implement in safe general way, and thus drop the ```HostingLogScope``` class create in ```RequestScope(...)```

```csharp
public abstract class HttpContext : ILogValues
{
    ...
   
    public override string ToString() => $"RequestId:{TraceIdentifier} RequestPath:{Request.Path}";

    IEnumerable<KeyValuePair<string, object>> ILogValues.GetValues()
    {
        yield return new KeyValuePair<string, object>("RequestId", TraceIdentifier);
        yield return new KeyValuePair<string, object>("RequestPath", Request.Path.ToString());
    }
}
```

Specific derived classes could then override with caching as per ```HostingLogScope```